app/code/Magento/Directory/Model/Currency/Import/CurrencyConverterApi.php

```diff
@@ -50,6 +55,11 @@ public function __construct(
      */
     public function fetchRates()
     {
+        $this->accessKey = $this->scopeConfig->getValue(
```

Comment:
Why use a member variable when you retrieve the value on every run anyways?

Reply:
I've referenced https://github.com/magento/magento2/blob/88c5485fcb7e85ad61d545712b0262b90dd4a0d8/app/code/Magento/Directory/Model/Currency/Import/FixerIo.php#L86-L88 for resolving this issue and found Magento is fetching access key in a loop, to avoid this problem I set this value into access key and use this member variable on other functions.   

What is your recommended method in this case?

Reply:
@Jitheesh just add a new parameter to the `convertBatch` method, since it's private it will be totally backward compatible. And you will avoid the adding of new member property.

Reply:
@sidolov I've added changes as per your comment.